src/kOS/Module/kOSVesselModule.cs

```diff
@@ -245,8 +254,51 @@ private void HookEvents()
         private void UnHookEvents()
         {
             ConnectivityManager.RemoveAutopilotHook(parentVessel, UpdateAutopilot);
+
+            if (Vessel.vesselType != VesselType.Unknown && Vessel.vesselType != VesselType.SpaceObject)
+            {
+                TimingManager.FixedUpdateRemove(TimingManager.TimingStage.ObscenelyEarly, cacheControllable);
+                TimingManager.FixedUpdateRemove(TimingManager.TimingStage.BetterLateThanNever, resetControllable);
+            }
         }
-        
+
+        #region Hack to fix "Require Signal for Control"
+        // TODO: Delete this hack if it ever gets fixed.
+
+        // Note: I am purposfully putting these variable declarations in this region instead of at the
+        // top of the file as our normal convention dictates.  These are specific to the hack and when
+        // we remove the hack the diff will make more sense if we wipe out a single contiguous block.
+        private bool workAroundControllable = false;
+        private CommNet.CommNetParams commNetParams;
+        private System.Reflection.FieldInfo isControllableField;
+
+        private void resetControllable()
+        {
+            if (workAroundControllable && isControllableField != null)
+            {
+                isControllableField.SetValue(parentVessel, false);
+            }
+        }
+
+        private void cacheControllable()
+        {
+            if (commNetParams == null)
+                commNetParams = HighLogic.CurrentGame.Parameters.CustomParams<CommNet.CommNetParams>();
+            if (!parentVessel.IsControllable && commNetParams != null && commNetParams.requireSignalForControl)
+            {
+                // HACK: Get around inability to affect throttle if connection is lost and require
+                if (isControllableField == null)
+                    isControllableField = parentVessel.GetType().GetField("isControllable", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+                isControllableField.SetValue(parentVessel, true);
+                workAroundControllable = true;
+            }
+            else
+            {
+                workAroundControllable = false;
+            }
+        }
+        #endregion
+
```

Comment:
Wow you're right this *IS* a nasty hack.  It depends on knowing exactly when stock chooses to run its checks for seeing if the vessel is controllable, so you can change it only when it's about to check, and then immediately put it back the way it was afterward.  If they ever do anything that fiddles with those timings it breaks.  

This looks VERY fragile.  I really hope they get this issue fixed for us in stock so we don't need to do something so invasive and can get rid of this as soon as possible.  But yeah, for now we sort of have no choice and have to do it.